Makefile

```diff
@@ -0,0 +1,32 @@
+export AWS_ACCESS_KEY_ID ?= test
+export AWS_SECRET_ACCESS_KEY ?= test
+export LOCAL_RUN ?= true
+SHELL := /bin/bash
+
+include .env
+
+build:
+		bin/build_lambdas.sh;
+
+awslocal-setup:
+		bin/deploy.sh
+
+terraform-setup:
+		$(MAKE) build
+		cd deployment/terraform; \
+		tflocal init; \
+		echo "Deploying Terraform configuration 🚀"; \
+		tflocal apply  --auto-approve -var="local_run=${LOCAL_RUN}"; \
+		echo "Paste the function URLs above to the WebApp 🎉";
+
+terraform-destroy:
+		cd deployment/terraform; \
+		tflocal destroy --auto-approve;
+
+start:
+		@LOCALSTACK_AUTH_TOKEN=$(LOCALSTACK_AUTH_TOKEN) localstack start -d
+
+stop:
+		localstack stop
+
+.PHONY: build awslocal-setup terraform-setup terraform-destroy start stop
```

Comment:
Thanks for adding `Makefile` to the sample 🥳 

We could enhance it further including the following targets as well: 
- add description for each target using say `##`.
    - For example:  `start:                    ## Start the localstack container in the detached mode`
- `usage`: 
    - display a help message showing each target in the `Makefile` that has a `##` comment next to it.
    - we can use the following steps do it:
       - searches through the Makefile for lines that contain `##`. 
       - modify the matched lines by removing everything after the target name and before the `##` comment.
       -  clean-up the output
       - sample output: 

```
usage                    Show this help
start                      Start the localstack container in the detached mode
```

- `install`: Installs the dependencies like localstack, terraform, etc. 
- `clean`: cleans the `.terraform` folder. 
- if the sample already deploys against AWS we can add the relevant targets for them as well

Please let me know if any help is required here 🙂

Reply:
I will add these targets. I have one question regarding,

> clean: cleans the .terraform folder.

what is these target for? why do we need to provide automation for this?

Reply:
Good point! It's not needed yet, it's just something we usually add to our makefiles.
Sometimes it can be confusing where state / build artifacts / ... are stored, then it's good to have a simple way to "clean everything" :)